packages/client/src/subscribe/ordering/OrderedMsgChain.ts

```diff
@@ -320,6 +323,11 @@ export class OrderedMsgChain {
                 }
             }
 
+            if (!this.dropSubsequentGapsOnFailedGapFill) {
```

Comment:
The `pop` method is called also when we receive data from realtime / resend? In that case we shouldn't set the counter to zero?